src/schemas/Clinical/index.ts

```diff
@@ -0,0 +1,454 @@
+      context: GlobalGqlContext,
+    ) => {
+      const { Authorization } = context;
+      const { shortName } = args;
+
+      const response = await clinicalService.getRegistrationData(shortName, Authorization);
+      return convertRegistrationDataToGql(shortName, { registration: response });
+    },
+    clinicalSubmissions: async (
+      obj: unknown,
+      args: { programShortName: string },
+      context: GlobalGqlContext,
+    ) => {
+      const { Authorization } = context;
+      const { programShortName } = args;
+
+      const response = await clinicalService.getClinicalSubmissionData(
+        programShortName,
+        Authorization,
+      );
+      return convertClinicalSubmissionDataToGql(programShortName, { submission: response });
+    },
+    clinicalSubmissionTypesList: async (obj: unknown, args: {}, context: GlobalGqlContext) => {
+      return await clinicalService.getClinicalSubmissionTypesList();
+    },
+    clinicalSubmissionSchemaVersion: async (obj: unknown, args: {}, context: GlobalGqlContext) => {
+      return await clinicalService.getClinicalSubmissionSchemaVersion();
+    },
+    clinicalSubmissionSystemDisabled: async (obj: unknown, args: {}, context: GlobalGqlContext) => {
+      return await clinicalService.getClinicalSubmissionSystemDisabled();
+    },
+  },
+  Mutation: {
+    uploadClinicalRegistration: async (
+      obj: unknown,
+      args: {
+        shortName: string;
+        registrationFile: FileUpload;
+      },
+      context: GlobalGqlContext,
+    ) => {
+      const { Authorization, egoToken } = context;
+      const { shortName, registrationFile } = args;
+      const permissions = egoTokenUtils.getPermissionsFromToken(egoToken);
+      // Here we are confirming that the user has at least some ability to write Program Data
+      // This is to reduce the opportunity for spamming the gateway with file uploads
+      if (!egoTokenUtils.canWriteSomeProgramData(permissions)) {
+        throw new AuthenticationError('User is not authorized to write data');
+      }
+
+      const { filename, createReadStream } = await registrationFile;
+      const fileStream = createReadStream();
+
+      // try {
+      const response = await clinicalService.uploadRegistrationData(
+        shortName,
+        filename,
+        fileStream,
+        Authorization,
+      );
+
+      return convertRegistrationDataToGql(shortName, response);
+    },
+    clearClinicalRegistration: async (
+      obj: unknown,
+      args: {
+        shortName: string;
+        registrationId: string;
+      },
+      context: GlobalGqlContext,
+    ) => {
+      const { Authorization } = context;
+      const { shortName, registrationId } = args;
+      const response = await clinicalService.clearRegistrationData(
+        shortName,
+        registrationId,
+        Authorization,
+      );
+      //@ts-ignore consult Jaser on modifying clearRegistrationData or remove this check altogether
+      if (response.error) {
```

Comment:
Took a look at the clinical code, there are some cases where response can be an error.... If the versionId doesn't match, submission doesn't exist, or if the submission is pending approval

Reply:
oops that submisison

Reply:
clear registration doesn't return anything if success, just code 200. I think the restErrorResponseHandler will take care of any other case where there is an error, but that won't make it to UI so yea I guess you could remove this.

Reply:
👍 cool thanks!